### PR TITLE
StoreCopier now warms up transformer i.e. allows it to invoke batched…

### PR DESCRIPTION
**Pull request by dharju**
… storeKeyConverter calls

### CODE REVIEW

File: ambry-api/src/main/java/com.github.ambry/store/Transformer.java [hunk 1]
```diff
@@ -26,6 +28,14 @@
    * @return the output {@link TransformationOutput}.
    */
   TransformationOutput transform(Message message);
+
+  /**
+   * Warms up transformer with message infos representing messages
+   * it will transform later
+   * @param messageInfos
```
Comment (vgkholla) [inline ambry-api/src/main/java/com.github.ambry/store/Transformer.java]: doc

File: ambry-api/src/main/java/com.github.ambry/store/Transformer.java [hunk 2]
```diff
@@ -26,6 +28,14 @@
...
+   * @throws Exception
```
Comment (vgkholla) [inline ambry-api/src/main/java/com.github.ambry/store/Transformer.java]: this is too broad. `transform()` doesn't throw an exception, I don't think this should either (runtime exceptions ok)

Comment (dharju) [reply ambry-api/src/main/java/com.github.ambry/store/Transformer.java]: So instead it should... return an Exception object?  Return a TransformationOutput that may have an Exception or an... empty Message object that means nothing?  idk seems like throwing an exception makes the most sense

File: ambry-api/src/main/java/com.github.ambry/store/Transformer.java (hunk 1 above)
Comment (dharju) [reply ambry-api/src/main/java/com.github.ambry/store/Transformer.java]: fixed

File: ambry-api/src/main/java/com.github.ambry/store/Transformer.java [hunk 3]
```diff
@@ -26,6 +28,15 @@
    * @return the output {@link TransformationOutput}.
    */
   TransformationOutput transform(Message message);
+
+  /**
+   * Warms up transformer with message infos representing messages
+   * it will transform later
+   * @param messageInfos message infos that will be used to warmup transformer,
+   *                     each mesage info
```
Comment (vgkholla) [inline ambry-api/src/main/java/com.github.ambry/store/Transformer.java]: incomplete?

Comment (dharju) [reply ambry-api/src/main/java/com.github.ambry/store/Transformer.java]: how's it incomplete? it's an interface so it's a generalized description

Comment (vgkholla) [reply ambry-api/src/main/java/com.github.ambry/store/Transformer.java]: "each message info"?

**Pull request merged**